src/ansys/fluent/core/quantity.py

```diff
@@ -532,13 +543,12 @@ def __init__(self, real_value, unit_str):
         self._value = float(real_value)
         self._unit = Unit(unit_str)
         self._dimension = Dimension(unit_str)
-        self._si_value = self._unit.si_factor * self._value
+        self._si_value = self._unit.si_factor * self._value + self._unit.si_offset
```

Comment:
We're also doing some specific stuff like this (in lambdas for temperature conversions). That other code still stands currently?

Reply:
Yes, it's a part of our main core `to` method and we have added above code to handle special case of `si_value` of degree Celsius and Fahrenheit.